src/core/mesh/qgsmeshdatasetgroupstore.h

```diff
@@ -0,0 +1,212 @@
+
+    //! Returns the metadata of the dataset with global \a index
+    QgsMeshDatasetMetadata datasetMetadata( const QgsMeshDatasetIndex &index ) const;
+
+    //! Returns the value of the dataset with global \a index and \a valueIndex
+    QgsMeshDatasetValue datasetValue( const QgsMeshDatasetIndex &index, int valueIndex ) const;
+
+    //! Returns \a count values of the dataset with global \a index and from \a valueIndex
+    QgsMeshDataBlock datasetValues( const QgsMeshDatasetIndex &index, int valueIndex, int count ) const;
+
+    //! Returns \a count 3D values of the dataset with global \a index and from \a valueIndex
+    QgsMesh3dDataBlock dataset3dValues( const QgsMeshDatasetIndex &index, int faceIndex, int count ) const;
+
+    //! Returns whether faces are active for particular dataset
+    QgsMeshDataBlock areFacesActive( const QgsMeshDatasetIndex &index, int faceIndex, int count ) const;
+
+    //! Returns whether face is active for particular dataset
+    bool isFaceActive( const QgsMeshDatasetIndex &index, int faceIndex ) const;
+
+    //! Returns the global dataset index of the dataset int the dataset group with \a groupIndex, corresponding to the relative \a time and the check \a method
+    QgsMeshDatasetIndex datasetIndexAtTime( quint64 time,
+                                            int groupIndex,
+                                            QgsMeshDataProviderTemporalCapabilities::MatchingTemporalDatasetMethod method ) const;
+
+    //! Returns the relative time of the dataset from the persistent provider reference time
+    quint64 datasetRelativeTime( const QgsMeshDatasetIndex &index ) const;
+
+    //! Returns wether at lea&st one of stored dataset group is temporal
+    bool hasTemporalCapabilities() const;
+
+    //! Write the store's information in a DOM document
+    QDomElement writeXml( QDomDocument &doc, const QgsReadWriteContext &context );
+
+    //! Read the store's information from a DOM document
+    void readXml( const QDomElement &storeElem, const QgsReadWriteContext &context );
+
+  signals:
+    //! emited after dataset groups are added
+    void datasetGroupsAdded( QList<int> indexes );
+
+  private slots:
+    void onPersistentDatasetAdded( int count );
+
+  private:
+
+    QgsMeshDataProvider *mPersistentProvider = nullptr;
+    std::unique_ptr<QgsMeshMemoryDatasetStore> mMemoryDatasets;
+    QMap < int, DatasetGroup> mRegistery;
+    std::unique_ptr<QgsMeshDatasetGroupTreeItem> mDatasetGroupTreeRootItem;
+
+    void removePersistentProvider();
+
+    DatasetGroup datasetGroup( int index ) const;
+    int newIndex();
+
+    int registerDatasetGroup( const DatasetGroup &group );
+    int nativeIndexToGroupIndex( QgsMeshDatasetSourceInterface *source, int providerIndex );
+    void createDatasetGroupTreeItems( const QList<int> &indexes );
+
+    //! Erase from the where this is store, not from the store (registery and tree item), for peristent dataset group, do nothing
```

Comment:
registery -> register
Erase -> Erases

Reply:
for regisyery, it is the noun not the verb.

Reply:
maybe, but unfortunately I do not understand at all the sentence